ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/SponsoredInteraction.java

```diff
@@ -0,0 +1,108 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.coders.*;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessageWithAttributesCoder;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Map;
+
+@AutoValue
+public abstract class SponsoredInteraction implements Serializable {
+    // topsites, suggest
+    public static final String SOURCE_TOPSITE = "topsites";
+    public static final String SOURCE_SUGGEST = "suggest";
+    abstract String source();
+
+    // desktop, phone, tablet
+    public static final String FORM_DESKTOP = "desktop";
+    public static final String FORM_PHONE = "phone";
+    public static final String FORM_TABLET = "tablet";
+    abstract String form();
+
+    // impression, click
+    public static final String INTERACTION_IMPRESSION = "impression";
+    public static final String INTERACTION_CLICK = "click";
+    abstract String interaction();
+
+    abstract String contextID();
+
+    @Nullable
+    abstract String reporterURL();
```

Comment:
I think this should be `reportingUrl` (with -ing) to be consistent with how it's named in the payload.